### ISSUE TITLE
Docker cp fails on concurrent copy for offline container

### ISSUE THREAD

**Issue by jialin-li** [vmware/vic#5742]
When two cp requests to an offline container are targeted at the same volume or r/w layer, one request would successfully attached the vmdk, and return from portlayer. When the second request comes in, the vmdk is already attached and mounted to the VCH, but the current logic failover to online and assumes the owner is a container VM. This is a problem because if the VCH starts a guest tool connection to itself, ref counting for mount is not updated. The vmdk will be unmounted whenever the first request finishes, and can cause the second copy operation to fail in the middle.

Acceptance:
- [ ] many (>20) concurrent copyTo operations to the same offline container succeed
- [ ] detection for images/volumes/containers for the case where the online Owner identified _is_ the applianceVM and appropriate retry logic (unsure how to test this).

**Comment by hickeng** [comment]
Working on dealing with the reference counting to handle the first part of the problem
Not yet dealing with scenario where the applianceVM is identified as the owner.

**Comment by matthewavery** [comment]
@hickeng you might confirm this. The reference counting issues were addressed in #5753 which is referenced above. The other side of the problem which George has detailed above is being fixed by #5780 it is still in review and needs tests. It is likely that I may be helping with that once I merge my current PR with passing tests.

**Comment by hickeng** [comment]
Yup - the two PRs related to this function are now merged into the feature branch. I've not added concurrency tests for cp as those were already present in @jialin-li 's PR.

I've also not added tests for the appliance being identified as the online owner. The only way I can think of to elicit this now I've closed the error paths is to have a container stopping just as a copy starts. It's a small timing hole there however and #5829 will close it further.
